cupy/core/_routines_indexing.pyx

```diff
@@ -49,6 +49,16 @@ cdef _ndarray_setitem(ndarray self, slices, value):
 
 
 cdef tuple _ndarray_nonzero(ndarray self):
+    cdef int ndim
+    cdef ndarray dst = _ndarray_argwhere(self)
+    ndim = len(dst)
+    if ndim > 1:
+        return tuple([dst[i] for i in range(ndim)])
```

Comment:
As `argwhere` is transposed relative to `nonzero`, you may want this to index along the right dimension. Perhaps fixes some of the testing issues you are seeing 🙂

```suggestion
        return tuple([dst[:, i] for i in range(ndim)])
```

Reply:
The `nonzero` is working perfectly in the case of `ndim > 1`,  because the argwhere is returning the transposed version. the traceback I commented above was of `argwhere`.

Reply:
This may now need to be updated since the kernel has changed. Though would be curious to hear what you find from your testing 🙂